teuthology/suite/run.py

```diff
@@ -345,31 +333,40 @@ def schedule_suite(self):
                 args=arg
             )
 
+            sha1 = self.base_config.sha1
             if config.suite_verify_ceph_hash:
                 full_job_config = dict()
                 deep_merge(full_job_config, self.base_config.to_dict())
                 deep_merge(full_job_config, parsed_yaml)
                 flavor = util.get_install_task_flavor(full_job_config)
-                sha1 = self.base_config.sha1
                 # Get package versions for this sha1, os_type and flavor. If
                 # we've already retrieved them in a previous loop, they'll be
                 # present in package_versions and gitbuilder will not be asked
                 # again for them.
-                package_versions = util.get_package_versions(
-                    sha1,
-                    os_type,
-                    flavor,
-                    package_versions
-                )
+                try:
+                    Run.package_versions = util.get_package_versions(
```

Comment:
The comment is wrong; I was thinking in general that "package version presence caching is global", but I was thinking "global across methods", which of course self would accomplish.  But since you mentioned it, using the class attr also makes it global across Run instances, which is still valid, although less useful because I don't think we ever instantiate more than one Run.

I can move it back into the instance.